src/qiboml/models/keras.py

```diff
@@ -33,29 +33,42 @@ def __post_init__(self):
 
         params = [p for param in self.circuit.get_parameters() for p in param]
         params = tf.Variable(self.backend.to_numpy(params))
-        self.circuit_parameters = self.add_weight(shape=params.shape, trainable=True)
-        self.set_weights([params])
+
+        self.circuit_parameters = self.add_weight(
+            shape=params.shape, initializer="random_normal", trainable=True
+        )
```

Comment:
You don't want to randomly initialized the parameters actually, but rather use the exact same parameters of the layers you built the model with, for this reason there was a `set_weights` in the first place.